aztec/src/main/kotlin/org/wordpress/aztec/AztecText.kt

```diff
@@ -257,6 +258,18 @@ class AztecText : AppCompatAutoCompleteTextView, TextWatcher, UnknownHtmlSpan.On
         enableTextChangedListener()
 
         isViewInitialized = true
+
+        viewTreeObserver.addOnScrollChangedListener {
+            if (this@AztecText.visibility == View.VISIBLE) {
+                if (invalidateMediaRunnable != null) {
+                    invalidateMediaHandler.removeCallbacks(invalidateMediaRunnable)
+                }
+                if (this@AztecText.text.getSpans(0, text.length, AztecMediaSpan::class.java).isNotEmpty()) {
+                    invalidateMediaRunnable = Runnable { this@AztecText.refreshText() }
+                    invalidateMediaHandler.postDelayed(invalidateMediaRunnable, 100L)
```

Comment:
Why are we refreshing text during scrolling? This causes the cursor to reset, the autocorrect rechecks whole text again, images will get reset too, which sometimes causes the scroll to"jump". Context menus are dismissed too.

Reply:
Yeah, sorry. It was a short and dirty way to call a redraw over all pictures.
We needed a way to call a redraw over all pictures in the editor. Pictures outside the bounds are set to null and the other redrawn. I guess there is a better way to do that (looping over all AztecMediaSpan??) Probably @0nko knows a better solution to the problem.

Reply:
How about just reseting spans? Ideally, we will only do it for spans that need it. Something like this:
```
editableText.getSpans(0, text.length, AztecMediaSpan::class.java).filter { it.isDirty }.forEach {
                    val spanStart = editableText.getSpanStart(it)
                    val spanEnd = editableText.getSpanEnd(it)
                    val flags = editableText.getSpanFlags(it)

                    invalidateMediaRunnable = Runnable { this@AztecText.editableText.setSpan(it, spanStart, spanEnd, flags) }
                    invalidateMediaHandler.postDelayed(invalidateMediaRunnable, 100L)
                }
```

Reply:
I've fixed it in https://github.com/wordpress-mobile/AztecEditor-Android/pull/474/commits/26b6b4d64b856218e633965426e68c2bfa110588 

Tested it and seems to work. Will let @0nko to run another round of testing with other devices.

Reply:
Unfortunately, this approach seems to be small jumps in spans when scrolling and it is most visible on the block quote:

![screenshot_1504774806](https://user-images.githubusercontent.com/1522856/30155038-1580c872-93bc-11e7-9d3b-a453d24e3083.png)

Reply:
I think it's because we are jamming all the span transactions into one Runnable. It works ok for me without it.

Reply:
I've tried different solutions but none of them completely fix the issues. 

- Setting the spans without using runnable isn't a viable solutions since the `OnScrollChangedListener` is called so many times, and there are a lot of redrawing involved while scrolling making it shuttering on lower devices or emulators.

- using 1 runnable for 1 span doesn't fix the issue. since the redraw is performed at the end of the scrolling and you will see the jump. Redraw is performed at the end of the scrolling since we remove pending runnable with `invalidateMediaHandler.removeCallbacks(null)`.
If we don't remove old callbacks the problem is then fixed but we're falling on performance issues like above, since too many runnables are created. 

I've then decided to use a "mixed" solution, re-set spans inline, each 10 cycle, and use runnable to make the final set/redraw. https://github.com/wordpress-mobile/AztecEditor-Android/commit/aa0942e052ec7c80f0ddd6e09962ec05ee3ffcf8

*Note that the problem is not in the image optimization code, I've disabled it and saw the jump on the first layout/redraw^. The underlying problem lies in the code below:
```
editableText.getSpans(0, text.length, AztecMediaSpan::class.java).forEach {
                                val spanStart = editableText.getSpanStart(it)
                                val spanEnd = editableText.getSpanEnd(it)
                                val flags = editableText.getSpanFlags(it)
                                editableText.setSpan(it, spanStart, spanEnd, flags)
                            }```

Don't know why for some strage reason some spans (it seems underline and quote in my case) change their dimensions. You can see this noticeable on the background box of the quote, that sometimes changes its size and there is 1px outside the correct position.


Anyway, this seems to be to most reliable and less hack-ish solution I came up with. We can change the cycle to something less than 10 if you still see the jump.